src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryText.cs

```diff
@@ -255,6 +257,76 @@ private static string ReplaceWith24HourClock(string format)
             return Regex.Replace(format, pattern, "HH${nonHours}", RegexOptions.IgnorePatternWhitespace);
         }
 
+        /// <summary>
+        /// Resolves month/minute ambiguity and 24h vs AM/PM.
+        /// </summary>
+        /// <param name="format">Format string.</param>
+        /// <returns></returns>
+        private static string ResolveAmbiguities(string format)
+        {
+            if (format.Length == 0)
+            {
+                return format;
+            }
+
+            // Handling especial cases
+            if (format.Length == 1)
+            {
+                // Cases not supported by Excel are not handled ie 'G'
+                var singleCharMatch = Regex.Match(format, "[dDhHmMsSyY]");
+
+                if (singleCharMatch.Success)
+                {
+                    switch (singleCharMatch.Value)
+                    {
+                        case "m":
+                        case "h":
+                            return $"%{singleCharMatch.Value.ToUpperInvariant()}";
+                        default:
+                            return $"%{singleCharMatch.Value}";
+                    }
+                    
+                }
+
+                return format;
+            }
+
+            var time12hMatch = Regex.Match(format, "[aA][mM]\\/[pP][mM]|[aA]\\/[pP]");
+
+            // Replace reserved chars
+            format = Regex.Replace(format, "[aA][mM]\\/[pP][mM]", "tt");
+            format = Regex.Replace(format, "[aA]\\/[pP]", "t");
+            format = Regex.Replace(format, "[hH]", time12hMatch.Success ? "h" : "H");
+            format = Regex.Replace(format, "[yY]", "y");
```

Comment:
>Regex.Replace(format, "[yY]", "y") [](http://example.com/codeflow?start=21&length=34)

`format.Replace('Y', 'y')` does the same, and is less expensive than using regex. Same with other similar cases below.